set/db.go

```diff
@@ -1186,27 +1185,41 @@ func (d *DB) doBackup() error {
 	return d.doRemoteBackup()
 }
 
-func (d *DB) doRemoteBackup() error {
+func (d *DB) doRemoteBackup() (err error) {
 	if d.remoteBackupPath == "" {
 		return nil
 	}
 
-	dir := filepath.Dir(d.remoteBackupPath)
-
-	err := d.remoteBackupHandler.EnsureCollection(dir)
+	putter, err := put.New(d.remoteBackupHandler, []*put.Request{{
+		Local:  d.backupPath,
+		Remote: d.remoteBackupPath,
+	}})
 	if err != nil {
 		return err
 	}
 
-	err = d.remoteBackupHandler.CollectionsDone()
-	if err != nil {
+	defer func() {
+		if errc := putter.Cleanup(); err == nil {
+			err = errc
+		}
+	}()
+
+	if err = putter.CreateCollections(); err != nil {
 		return err
 	}
 
-	return d.remoteBackupHandler.Put(&put.Request{
-		Local:  d.backupPath,
-		Remote: d.remoteBackupPath,
-	})
+	started, finished, skipped := putter.Put()
+
+	drainChannel(started)
+	drainChannel(finished)
+	drainChannel(skipped)
+
+	return err
+}
+
+func drainChannel(ch chan *put.Request) {
+	for range ch {
```

Comment:
Does one of these work and get rid of the linting error?

```
for range ch {}
```

```
for len(ch) > 0 {
  <-ch
}
```

Reply:
Gofmt undoes the first, and the second seems racey to me; we could do the len check before anything is on the channel.

Reply:
The more complicated way to drain a channel would be:
```
	for {
		select {
		case _, ok := <-ch:
			if !ok {
				return
			}
		}
	}
```

However, thart produces a local linting error:
`should use for range instead of for { select {} }`

Reply:
Ok, go back to the original and add a nolint comment.